crates/subspace-farmer/src/commitments.rs

```diff
@@ -216,16 +216,37 @@ impl Commitments {
     }
 
     /// Finds the commitment/s falling in the range of the challenge
-    pub(crate) fn find_by_range(&self, target: Tag, range: u64, salt: Salt) -> Option<(Tag, u64)> {
+    pub(crate) fn find_by_range(
+        &self,
+        target: Tag,
+        range: u64,
+        salt: Salt,
+    ) -> Option<(Tag, PieceIndex)> {
+        fn iter_solutions(
+            iter: &mut rocksdb::DBRawIteratorWithThreadMode<'_, DB>,
+        ) -> Option<(Tag, PieceIndex)> {
+            match (iter.key(), iter.value()) {
+                (Some(key), Some(value)) => {
+                    let result = Some((
+                        key.try_into().unwrap(),
+                        PieceIndex::from_be_bytes(value.try_into().unwrap()),
+                    ));
+
+                    iter.next();
+                    result
+                }
+                (None, None) => None,
+                _ => unreachable!("RockDB should always return value if key is present"),
+            }
+        }
```

Comment:
I appreciate the desire to reduce code duplication, but previous code was simple and boring, new code is more complex to understand and is even longer in the end.